### PR TITLE
140038 - Bess Builder Service AD authorization for Ess and Fss API

### PR DESCRIPTION
**Pull request by rupali15127**
This PR consists changes of [PBI 140038](https://dev.azure.com/ukhydro/Exchange%20Set%20Service/_workitems/edit/140038) and [PBI 140032](https://dev.azure.com/ukhydro/Exchange%20Set%20Service/_workitems/edit/140032)

- Code changes for ad auth for ess and fss api
- Unit test cases 
- Mocks related changes in FSS and ESS for BESS
- Added functional tests

### CODE REVIEW

File: UKHO.PeriodicOutputService/UKHO.BESS.API.FunctionalTests/FunctionalTests/EssIntegration/EssIntegrationTests.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            HttpResponseMessage apiResponse = await EssEndpointHelper.ProductVersionsEndpoint(testConfiguration.authTokenConfig.BaseUrl, productVersionData, "s63", true);
+            apiResponse.StatusCode.Should().Be((HttpStatusCode)200);
+        }
+
+        //Product Backlog Item 140038: BESS BS - AD auth for ESS and FSS API
+        [Test]
+        public async Task WhenICallEssProductVersionEndpointWithIncorrectURL_ThenBadRequestStatusCode400IsReturned()
+        {
+            productVersionData = new List<ProductVersionModel> {
+            dataHelper.GetProductVersionData("GB301910", 1, 0)
```
Comment (rsj-github) [inline UKHO.PeriodicOutputService/UKHO.BESS.API.FunctionalTests/FunctionalTests/EssIntegration/EssIntegrationTests.cs:57]: hard coding

**Review by rsj-github — CHANGES_REQUESTED** [review]
in general hard coding isn't best practice and leads to fragility

File: UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Controllers/ExchangeSetServiceController.cs [hunk 2]
```diff
@@ -16,48 +16,72 @@ public ExchangeSetServiceController(ExchangeSetService exchangeSetService)
...
         {
             if (productIdentifiers != null && productIdentifiers.Any())
             {
-                ExchangeSetServiceResponse? response = _exchangeSetService.CreateExchangeSetForPostProductIdentifier(productIdentifiers);
-                if (response == null)
+                if (!string.IsNullOrEmpty(exchangeSetStandard))
                 {
-                    return BadRequest();
+                    exchangeSetStandard = exchangeSetStandard.ToLower();
+                }
```
[the remark this answers is not in the document]
Comment (shirinbt14926) [reply UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Controllers/ExchangeSetServiceController.cs]: Same as previous comment

File: UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Controllers/ExchangeSetServiceController.cs [hunk 3]
```diff
@@ -16,48 +16,72 @@ public ExchangeSetServiceController(ExchangeSetService exchangeSetService)
...
+        public IActionResult PostProductVersions([FromBody] List<ProductVersionRequest> productVersionRequest, [FromQuery] string? exchangeSetStandard)
         {
             if (productVersionRequest != null && productVersionRequest.Any())
             {
-                ExchangeSetServiceResponse? response = _exchangeSetService.CreateExchangeSetForPostProductVersion(productVersionRequest);
-                if (response == null)
+                if (!string.IsNullOrEmpty(exchangeSetStandard))
+                {
+                    exchangeSetStandard = exchangeSetStandard.ToLower();
+                }
```
[the remark this answers is not in the document]
Comment (shirinbt14926) [reply UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Controllers/ExchangeSetServiceController.cs:76]: Same as previous comment

File: UKHO.PeriodicOutputService/UKHO.BESS.API.FunctionalTests/FunctionalTests/EssIntegration/EssIntegrationTests.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+using UKHO.BESS.API.FunctionalTests.Helpers;
+using UKHO.BESS.API.FunctionalTests.Models;
+
+namespace UKHO.BESS.API.FunctionalTests.FunctionalTests.EssIntegration
+{
+    public class EssIntegrationTests
+    {
+        static readonly TestConfiguration testConfiguration = new();
+        public DataHelper dataHelper = new();
+        protected List<string> productIdentifiers = new() { "GB301910" };
```
[the remark this answers is not in the document]
Comment (Mayuresh-Gawde) [reply UKHO.PeriodicOutputService/UKHO.BESS.API.FunctionalTests/FunctionalTests/EssIntegration/EssIntegrationTests.cs]: Resolved!!

File: UKHO.PeriodicOutputService/UKHO.BESS.API.FunctionalTests/FunctionalTests/EssIntegration/EssIntegrationTests.cs (hunk 1 above)
Comment (Mayuresh-Gawde) [reply UKHO.PeriodicOutputService/UKHO.BESS.API.FunctionalTests/FunctionalTests/EssIntegration/EssIntegrationTests.cs]: Resolved!!

File: UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Services/ExchangeSetService.cs [hunk 5]
```diff
@@ -53,9 +54,10 @@ public ExchangeSetServiceResponse CreateExchangeSetForGetProductDataSinceDateTim
...
 
 
-        public ExchangeSetServiceResponse CreateExchangeSetForPostProductIdentifier(string[] productIdentifiers)
+        public ExchangeSetServiceResponse CreateExchangeSetForPostProductIdentifier(string[] productIdentifiers, string exchangeSetStandard)
         {
             string productIdentifiersPattern = "productIdentifier-" + string.Join("-", productIdentifiers);
+            productIdentifiersPattern += !string.IsNullOrEmpty(exchangeSetStandard) ? "-" + exchangeSetStandard : "";
             CreateBatchRequest batchRequest;
 
             if (productIdentifiers.Contains("GB800001") ||
```
[the remark this answers is not in the document]
Comment (harshal11869) [reply UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Services/ExchangeSetService.cs:63]: This is mock service and mainly this hardcoded cells differentiate the request is for AIO or No AIO and it also get utilized by both POS & POS AIO services.

File: UKHO.PeriodicOutputService/UKHO.BESS.BuilderService/appsettings.json [hunk 6]
```diff
@@ -25,12 +25,23 @@
         "Environment": "dev",
         "EntityPath": "",
         "System": "BESS Webjob",
-        "Service": "BESS Builder Service",
+        "Service": "BESS Configuration Service",
```
Comment (Nadeem14456) [inline UKHO.PeriodicOutputService/UKHO.BESS.BuilderService/appsettings.json:28]:
```suggestion
        "Service": "BESS Builder Service",
```

**Review by harshal11869 — APPROVED** [review]
Looks Good.

**Review by rsj-github — APPROVED** [review]

**Review by JiviteshT — APPROVED** [review]
Looks good

**Pull request merged**